utils/helper.js

```
@@ -0,0 +1,185 @@
+  return unwrapRecurse(initialValue, [], callback);
+};
+
+const map = function (array, callback) {
+  return reduce(
+    array,
+    (previousValue, currentValue, index) => {
+      return [...previousValue, callback(currentValue, index, array)];
+    },
+    []
+  );
+};
+
+const filter = function (array, callback) {
+  return reduce(
+    array,
+    function (previousValue, currentValue, index) {
+      return callback(currentValue, index, array) ? [...previousValue, currentValue] :
+        previousValue;
+    },
+    []
+  );
+};
+
+const avgEven = function (array, paramFilter = filter, paramReduce = reduce) {
+  const evenArray = paramFilter(array, item => (item % 2 == 0 ? true : false));
+  if (!evenArray.length) {
+    return undefined;
+  }
+
+  const sumEvenArray = paramReduce(
+    evenArray,
+    (previous, current) => previous + current
+  );
+
+  return sumEvenArray / evenArray.length;
+};
+
+const randSum = function () {
+  let count = 0;
+  const array = unwrap(
+    current =>
+    ++count >= 10 ?
+    false : [current, current + Math.floor(Math.random() * 10)],
+    1
+  );
+
+  return reduce(
+    array,
+    (previousValue, currentValue) => previousValue + currentValue
+  );
+};
+
+const findFirstRecurse = function (currentValue, index, array, callback) {
+  if (currentValue === undefined) {
+    return undefined;
+  }
+  if (!callback(currentValue, index, array)) {
+    return findFirstRecurse(array[++index], index, array, callback);
+  } else {
+    return array[index];
+  }
+};
+
+const first = function (array, callback) {
+  return findFirstRecurse(array[0], 0, array, callback);
+};
+
+const lazy = function (callback, ...param) {
+  let res;
+  let isCalled = false;
+
+  return function () {
+    if (!isCalled) {
+      isCalled = true;
+      return (res = callback(...param));
+    } else {
+      return res;
+    }
+  };
+};
+
+const memoize = function (callback) {
+  const cache = {};
+  return function (param) {
+    try {
```

You better not wrap your entire function in try.. catch. It's bad practice, because it can hide errors in your code